sql/src/test/java/io/crate/operation/collect/ShardCollectServiceTest.java

```diff
@@ -45,41 +46,54 @@
 
 import static org.hamcrest.Matchers.is;
 
-@ESIntegTestCase.ClusterScope(numDataNodes = 1)
+@ESIntegTestCase.ClusterScope(numDataNodes = 1, numClientNodes = 0)
 public class ShardCollectServiceTest extends SQLHttpIntegrationTest {
 
     @Test
     public void testReadIsolation() throws Exception {
         execute("create blob table b1 clustered into 1 shards with (number_of_replicas = 0)");
-        ensureGreen();
         upload("b1", "foo");
         upload("b1", "bar");
+        ensureGreen();
+
+        assertBusy(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    IndexService indexService = internalCluster().getInstance(IndicesService.class).indexService(".blob_b1");
+                    assertNotNull(indexService);
```

Comment:
I wanted to make sure that I get the shardCollectService.
So I would need to define the vars (indexService, etc) before assert busy and then 
create a private class with constructor that updates them and pass it as runnable).
I prefered to use the anonymous class but then everything must be inside.

Any better suggestion?